lib/net.js

```diff
@@ -899,24 +899,18 @@ function connect(self, address, port, addressType, localAddress, localPort) {
 }
 
 
-Socket.prototype.connect = function(options, cb) {
+Socket.prototype.connect = function() {
+  const args = new Array(arguments.length);
+  for (var i = 0; i < arguments.length; i++)
+    args[i] = arguments[i];
+  // TODO(joyeecheung): use destructuring when V8 is fast enough
```

Comment:
@vhain I don't think it *needs* to be reverted if no one objects to / spots a bug caused by moving it down though.